pkg/k8s/watchers/service.go

```diff
@@ -104,5 +107,8 @@ func (k *K8sWatcher) deleteK8sServiceV1(svc *slim_corev1.Service, swg *lock.Stop
 			k.redirectPolicyManager.OnDeleteService(svcID)
 		}
 	}
+	if option.Config.BGPAnnounceLBIP {
+		k.bgpSpeakerManager.OnDeleteService(svc)
```

Comment:
I'd much prefer if all of the dedicated BGP handling logic was under one package. In general what that would mean is during initialization, any other logic that wants to learn about various events should subscribe into the watcher, then in the watcher we would just iterate through all of the things that need the service updates.

Given that we're already leaking such details into the k8s watchers and it's just a couple of cases so far I'm inclined to let this slide in this case but we should probably start thinking about how to keep the core logic free of all of these extra conditionals and references to various features.

Reply:
^ Looks like you already raised this concern in the PR description as a concern. :+1:

Reply:
Yes, this will be addressed in https://github.com/cilium/cilium/issues/15471.